addons/roe/roe.lua

```diff
@@ -0,0 +1,280 @@
+    complete = T{}    
+}
+
+
+local function cancel_roe(id)
+    id = tonumber(id)
+    
+    if not id then return end
+    
+    if settings.blacklist[id] then return end
+    
+    local p = packets.new('outgoing', 0x10d, {['RoE Quest'] = id })
+    packets.inject(p)
+end
+
+local function accept_roe(id)
+    id = tonumber(id)
+    
+    if not id or _roe.complete[id] or _roe.active[id] then return end
+    
+    local p = packets.new('outgoing', 0x10c, {['RoE Quest'] = id })
+    packets.inject(p)
+end
+
+local function eval(...)
+    assert(loadstring(table.concat({...}, ' ')))()
+end
+
+local function save(name)
+    if not type(name) == "string" then
+        error('`save` : specify a profile name')
+        return
+    end
+    
+    name = name:lower()
+
+    settings.profiles[name] = S(_roe.active:keyset())
+    settings:save('global')
+    notice('saved %d objectives to the profile %s':format(_roe.active:length(), name))
+end
+
+
+local function list()
+    notice('You have saved the following profiles: ')
+    notice(settings.profiles:keyset())
+end
+
+local function set(name)
+    if not type(name) == "string" then
+        error('`set` : specify a profile name')
+        return
+    end
+    
+    name = name:lower()
+    
+    if not settings.profiles[name] then
+        error('`set` : the profile \'%s\' does not exist':format(name))
+        return
+    end
+    
+    if type(settings.profiles[name]) == "string" then
+        settings.profiles[name] = S(settings.profiles[name]:split(','):map(tonumber))
+    end
+    
+
+    
+    local needed_quests = settings.profiles[name]:diff(_roe.active:keyset())   
+    local available_slots = 30 - _roe.active:length()
+    local to_remove = S{}
+           
+    if settings.clearall then
+        to_remove:update(_roe.active:keyset())
+    elseif settings.clear then
+        for id,progress in pairs(_roe.active) do
+            if (needed_quests:length() - to_remove:length()) <= available_slots then
+                break
+            end
+            if (progress == 0 or settings.clearprogress) and not settings.blacklist[id] then
+                to_remove:add(id)
+            end
+        end
+    end
+            
+        
+    if (needed_quests:length() - to_remove:length()) > available_slots then
+        error('you do not have enough available quest slots')
+        return
+    end
+    
+    for id in to_remove:it() do
+        cancel_roe(id)
+        coroutine.sleep(.5)
+    end
+    
+    for id in needed_quests:it() do
+        accept_roe(id)
+        coroutine.sleep(.5)
+    end
+    
+    notice('loaded the profile \'%s\'':format(name))
+
+end
+
+local function unset()
+    for id,progress in pairs(_roe.active:copy()) do
+        if progress == 0 or settings.clearprogress then
+            cancel_roe(id)
+            coroutine.sleep(.5)
+        end
+    end
+end
+
+local true_strings = S{'true','t','y','yes','on'}
+local false_strings = S{'false','f','n','off'}

```

Comment:
`'no'` is missing here, unless that was intentional.

Reply:
It was not!